File: Data Structures & Algorithms/longest-consecutive-sequence/submission-0.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& n) {
        // [0],[3],[7],[2],[5],[8],[4],[6],[0],[1]
        // 0, 0, 1, 2, 3, 4, 5, 6, 7, 8
        int s = n.size();
        sort(n.begin(), n.end());
        // 1 2 3 4 100 200
        int l{1};
        int j{1};
        if(!s){return 0;}
        for(int i{1}; i<s; i++){
            if((n[i]-n[i-1])==1){
                j++;
            }
            else if(n[i]==n[i-1]){continue;}
            else { l = max(j,l); j = 1; }
        }

        return max(j,l);
    }
};

File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-0.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def dfs(self, node, Max):
        if not node:
            return 0
        good = 0
        if(node.val >= Max):
            Max = max(node.val, Max)
            good = 1
        good += self.dfs(node.right, Max) + self.dfs(node.left, Max)
        return good 

    def goodNodes(self, root: TreeNode) -> int:
        return self.dfs(root, root.val)